nixos/modules/services/networking/nix-store-gcs-proxy.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  opts = { name, config, ... }: {
+    options = {
+      enable = mkOption {
+        default = true;
+        type = types.bool;
+        example = true;
+        description = "Whether to enable proxy for this bucket";
+      };
+      bucketName = mkOption {
+        type = types.str;
+        default = name;
+        example = "my-bucket-name";
+        description = "Name of Google storage bucket";
+      };
+      address = mkOption {
+        type = types.str;
+        example = "localhost:3000";
+        description = "The address of the proxy.";
+      };
+    };
+  };
+  enabledProxies = lib.filterAttrs (n: v: v.enable) config.services.nix-store-gcs-proxy;
+  mapProxies = function: lib.mkMerge (lib.mapAttrsToList function enabledProxies);
+in
+{
+  options.services.nix-store-gcs-proxy = mkOption {
+    type = types.attrsOf (types.submodule opts);
+    default = {};
+    description = ''
+      Attribute set for a proxy for a particular GCS bucket.
+    '';
+  };
+
+  config.systemd.services = mapProxies (name: cfg: {
```

Comment:
I am not sure of the implications this has on the eval of nixpkgs. Since we are always mapping (e.g. an empty list) of attributes here this has to be done for all and every NixOS out there.

Usually that is guarded by an `mkIf (buckets ! =[])` or similar.

Reply:
Very negligible for performance, I wouldn't worry about this.